### DISCUSSION TITLE
Submarine editor improvements

### DISCUSSION THREAD

**Discussion by rav2n** [FakeFishGames/Barotrauma#4736] [Ideas and suggestions]
1. Add the option to see the parameters of the submarine without upgrades and with the max improved characteristics. This is really lacking, especially when it comes to speed.
2. Ruler and option to add guidelines. Not really necessary, but why not?
3. Ability to create controlled spawn points for creatures. (spawn conditions, monster type, spawn probability.) This will greatly expand the possibilities when creating custom outposts, wrecks... alien ruins?
4. Ability to turn off the backlight when hovering over an object. Example: I want to make an interesting looking engine, for this I take one of the available ones and make it transparent. Then I sculpt on top of it something that my imagination and perseverance are enough for. Then, already in the game, I hover over the object and see the contours of the original engine... This makes me sad.
5. Rotation of walls and background. Will greatly expand the capabilities of sub-builders. As far as I know, this problem was partially solved by modmakers, but I would still like a solution from the developers.

**Comment by 3e849f2e5c** [comment]
> Ruler and option to add guidelines. Not really necessary, but why not?

Could you explain in more detail how you'd like to see this implemented? The next version will add a global grid and a "measuring tape" that can be accessed by holding Alt but I imagine you have something else in mind and I'm not a mind reader

![](https://kosblue.com/e/kznrj.png)

**Reply by rav2n** [reply]
I meant something like that in photoshop. A ruler with divisions in height and length, with the ability to create guidelines for aligning elements. But what you implement looks great too! I just didn't know about it.

**Reply by Rokvach** [reply]
You meant this type of ruler which takes in the cursor position and lets you place down guidelines in either X or Y axis globally?
![image](https://user-images.githubusercontent.com/7299826/104591439-5b6d8a80-5675-11eb-878c-c277b6b5e1f7.png)

**Reply by rav2n** [reply]
Yes

**Reply by 3e849f2e5c** [reply]
I like the idea and it would be easy to implement but it wouldn't be easy to make persistent unfortunately, just a matter of figuring out where to put the rulers on the screen as top and bottom are occupied

**Reply by Rokvach** [reply]
On photoshop its in the "viewport", needs to be toggled on to see it
![image](https://user-images.githubusercontent.com/7299826/104599296-3fbbb180-5680-11eb-936b-840f5c7b3eb8.png)

**Reply by MaaiG** [reply]
> I like the idea and it would be easy to implement but it wouldn't be easy to make persistent unfortunately, just a matter of figuring out where to put the rulers on the screen as top and bottom are occupied

Well, something like that (And it would be good if there were Photoshopped guidance lines!):
![Безымянный-1](https://user-images.githubusercontent.com/93348499/145634471-a5fc9801-f8a4-4e23-b55f-814a47b6a2b8.gif)
What do you think? Is there any chance of getting that kind of tool?

**Comment by Konfuz** [comment]
> Rotation of walls and background. Will greatly expand the capabilities of sub-builders. As far as I know, this problem was partially solved by modmakers, but I would still like a solution from the developers.

I recon this might be a limitation of the engine. I proposed an improvement to clip textures in order to support making backgrounds for roundish room borders (such as the curved window) but it was rejected on grounds of the renderer being unable at this point: #4599 

The engine however *does* support textures alpha channel. A great improvement would be to be able to *paint* alpha in the editor on the background like you do on the dirt-layer when you use an ethanol filled spraygun ingame. That way the renderer does not need to be touched. You only make an existing function accessible in a different context - and the result would be even more powerful than my suggestion in #4599 allowing not only to adapt the (perceived) shape of the texture border, but also for "holes" inside the texture, spots of discoloration, manual blending of texture borders etc. This should eliminate all instances where you have to "stich" together a sawtooth-pattern of multiple textures in order to match a curved hull.
